### ISSUE TITLE
Predicting Same label for every data in custom dataset

### ISSUE THREAD

**Issue by mdshopon** [cszhangzhen/HGP-SL#3]
Thank you very much for thi implementation. It was really helpful 

Currently, I am training a dataset of my own, for human activity recognition. I have converted the skeleton data into graph structure.

Number of class = 75 
Number of features = 4

edge_index is always same for every graph. 

While training, my model is always predicting the same label for every data that's why loss is not decreasing and accuracy is not increasing. Is there any modifications I further need to do to run?

**Comment by cszhangzhen** [comment]
Hi,

"edge_index is always same for every graph." Did you mean the graph structure is always the same for each graph data?

How large is your dataset? Maybe you can provide more information.

**Comment by mdshopon** [comment]
Yes, the graph structure is always same for every data. 
I have total 7000 training samples.

**Comment by cszhangzhen** [comment]
Currently, I have no idea why it fails to work on your custom datasets.

Maybe you can tune the hyper-parameters like embedding size, pooling ratio etc.

Another way to debug this problem is to use a more simple module. For example, you can use GCN and global mean pooling to perform graph classification. If this simple solution does not work either, it might be caused by your custom datasets.

**Comment by mdshopon** [comment]
Thank you for your suggestion. I tried tuning hyperparameters but didn't work. I even tried to use a very simple GCN architecture. But the result is still same. 

Is it a problem if the graph structure is same for all data? 

My feature values are the co ordinate points of skeleton joints. I checked the dataset thoroughly and so far I found no issues with the dataset. Any suggestion?

**Comment by cszhangzhen** [comment]
Hi,

Traditional graph data usually has different structure. Since your data is human skeleton data, the graph structure is same for all data which is reasonable. 

As far as I know, in human activity recognition task, you should take a sequence of graphs as input and model their temporal structure changes for action recognition, instead of classifying each single graph data.
